src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -84,6 +96,38 @@ public String updateUserEmail(UserChangeEmailDto userChangeEmailDto, User user)
         return getMessageSource("changed.user.email");
     }
 
+    public String deleteUserFirst(User user) {
+        user.setStatus(DELETED);
+        userRepository.saveAndFlush(user);
+
+        return getParametrizedMessageSource("account.deleted.first", getDaysBeforeDeletion(user));
+    }
+
+    public long getDaysBeforeDeletion(User user) {
+        return numberOfDaysToKeepDeletedUsers - (DAYS.between(user.getUpdated(), LocalDateTime.now()));
+    }
+
+    @Scheduled(cron = ONE_TIME_PER_DAY_AT_3_AM)
+    public void permanentlyDeleteUsers() {
+        userRepository.findAll().stream()
+                .filter(user -> user.getStatus().equals(DELETED) &&
+                        isDurationMoreThanNumberOfDaysToKeepDeletedUser(user.getUpdated()))
+                .forEach(userRepository::delete);
+    }
+
+    private boolean isDurationMoreThanNumberOfDaysToKeepDeletedUser(LocalDateTime localDateTime) {
+        Duration duration = Duration.between(localDateTime, LocalDateTime.now());
+
+        return duration.toDays() > numberOfDaysToKeepDeletedUsers;
+    }
+
+    public String restoreUser(User user) {
```

Comment:
You should change the name because you don't restore the user here but make Active again.